php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java

```diff
@@ -1604,6 +1608,45 @@ public boolean isAccepted(PhpElement element) {
         }
     }
 
+    private void autoCompleteConstructorParameterName(final PHPCompletionResult completionResult, final PHPCompletionItem.CompletionRequest request) {
+        if (CancelSupport.getDefault().isCancelled()) {
+            return;
+        }
+        TokenHierarchy<?> th = request.info.getSnapshot().getTokenHierarchy();
+        TokenSequence<PHPTokenId> tokenSequence = LexUtilities.getPHPTokenSequence(th, request.anchor);
+        if (tokenSequence == null) {
+            return;
+        }
+        Token<? extends PHPTokenId> constructorTypeName = CompletionContextFinder.findFunctionInvocationName(tokenSequence, request.anchor);
+        if (constructorTypeName != null) {
+            Model model = request.result.getModel();
+            NamespaceScope namespaceScope = ModelUtils.getNamespaceScope(model.getFileScope(), request.anchor);
+            String fqTypeName = VariousUtils.qualifyTypeNames(constructorTypeName.text().toString(), request.anchor, namespaceScope);
+            Set<AliasedName> aliasedNames = ModelUtils.getAliasedNames(model, request.anchor);
+            Set<MethodElement> constructors = request.index.getConstructors(NameKind.exact(fqTypeName), aliasedNames, Trait.ALIAS);
+            Set<String> duplicateCheck = new HashSet<>();
+            for (MethodElement constructor : constructors) {
+                if (CancelSupport.getDefault().isCancelled()) {
+                    return;
+                }
+                for (ParameterElement parameter : constructor.getParameters()) {
+                    if (CancelSupport.getDefault().isCancelled()) {
+                        return;
+                    }
+                    String name = parameter.getName();
+                    if (name != null) {
```

Comment:
Not sure, but maybe, to be safe, we should also check the length of the string?

Reply:
Yes, we should check it. Thanks!

Reply:
```suggestion
                    if (!StringUtils.isEmpty(name)) {
```
Is this OK because check `name.length() > 0`?

Reply:
Fixed: https://github.com/apache/netbeans/compare/1056bc237e4a92e745d148690f04ab1210b7a6ae..24ba6df45f395f1cbec9459526befa6f48345de9

I've introduced a method because there are duplicated parts.

Reply:
Looks good.